tests/test_8schools.py

```diff
@@ -18,9 +21,51 @@ def schools_pm4():
     return obs
 
 
+def test_model_logp():
+    """
+    Recreate this with
+
+    ```python
+    import pymc3 as pm
+    import numpy as np
+
+    J = 8
+    y = np.array([28, 8, -3, 7, -1, 1, 18, 12])
+    sigma = np.array([15, 10, 16, 11, 9, 11, 10, 18])
+
+
+    with pm.Model() as eight_schools:
+        eta = pm.Normal("eta", 0, 1, shape=J)
+        mu = pm.Normal("mu", 1, 10)
+        tau = pm.HalfNormal("tau", 2.0)
+
+        theta = mu + tau * eta
+
+        pm.Normal("obs", theta, sigma, observed=y)
+
+    print(eight_schools.logp({'eta': np.zeros(8), 'mu': 0, 'tau_log__': 1}).astype(np.float32))
+    ```
+    """
+    logp, *_ = pm4.inference.sampling.build_logp_and_deterministic_functions(
+        schools_pm4(), observed={"obs": y}, state=None
+    )
+    init_value = logp(
+        **{
+            "schools_pm4/eta": tf.zeros(8),
+            "schools_pm4/mu": tf.zeros(()),
+            "schools_pm4/__log_tau": tf.ones(()),
+        }
+    ).numpy()
+
+    assert_almost_equal(init_value, -42.876_114)
```

Comment:
note this new test makes sure that the log probability is correct (for at least one point!)